### ISSUE TITLE
API - Create new alias, is there any post rate limit?

### ISSUE THREAD

**Issue by highend** [simple-login/app#127]
Hi,

just a quick question before I try to "mass-create" aliases (I need several hundred):
Do I need to add any artificial wait timeout between POST /api/alias/custom/new or can I send them "as fast as possible"?

**Comment by nguyenkims** [comment]
Hey it’s better to put some « sleep » in between as otherwise your IP might be blacklisted by our server. Sleep for 1sec every 5 aliases should be enough.

**Comment by highend** [comment]
Hi,

ok, I'll do as suggested. Thanks!

**Comment by jeifour** [comment]
@nguyenkims Has the rate limit changed? I wrote a script that is supposed to create around 100 aliases. I put 2 second sleep after every single API request. After 5 requests I got `Rate limit exceeded`.

**Comment by nguyenkims** [comment]
@jeifour we have a recent change in our rate limit algorithm which might explain the error. Is the error still there when you increase the sleep in between?

**Comment by jeifour** [comment]
@nguyenkims I have reviewed the code again here. It seems like it was this rate limit which I was running into:

https://github.com/simple-login/app/blob/f47661c3d26de7fa51a7f3a44ecbaa185a182493/example.env#L169

For my purposes the hourly limit was a bit low. But I got it working fine. Now that the CSV batch import got fixed for me in https://github.com/simple-login/app/pull/1268/commits/86e96aef290c0cee1260021b5f248db9d20c97f9 (thanks btw, that was fast!) next time I do any bigger change with my aliases I can use that feature again 😄
